src/backend/commands/matview.c

```diff
@@ -1526,34 +1534,39 @@ IVM_immediate_maintenance(PG_FUNCTION_ARGS)
 	/*
 	 * Switch to the owner's userid, so that any functions are run as that
 	 * user.  Also arrange to make GUC variable changes local to this command.
-	 * Don't lock it down too tight to create a temporary table just yet.  We
-	 * will switch modes when we are about to execute user code.
+	 * We will switch modes when we are about to execute user code.
 	 */
 	relowner = matviewRel->rd_rel->relowner;
 	GetUserIdAndSecContext(&save_userid, &save_sec_context);
 	SetUserIdAndSecContext(relowner,
 						   save_sec_context | SECURITY_LOCAL_USERID_CHANGE);
```

Comment:
Changing user with "ECURITY_LOCAL_USERID_CHANGE" was required for creating temptables, otherwise DefineRelation() raises an error:

```
  642     if (stmt->relation->relpersistence == RELPERSISTENCE_TEMP
  643         && InSecurityRestrictedOperation())
  644         ereport(ERROR,
  645                 (errcode(ERRCODE_INSUFFICIENT_PRIVILEGE),
  646                  errmsg("cannot create temporary table within security-restricted operation")));
```

Now that we don't create temptables, so I think we don't need to use SECURITY_LOCAL_USERID_CHANGE, and we can lock down security-restricted operations by using SECURITY_RESTRICTED_OPERATION at this timing.  Also, the comment will need modify.

Reply:
I just thought so too. maybe, we don't need to get locks dived into two.

Reply:
Yes, we can remove unnecessary codes not for confusing someone who read it.

Reply:
I agree with you. first I fixed this

Reply:
Thanks. It seems good.